File: solutions/problem_2206_divide_array_into_equal_pairs/solution.cpp
/*
 * Problem 2206: Divide Array Into Equal Pairs
 * =========================================
 * Difficulty: Easy
 * Tags: Array, Hash Table, Bit Manipulation, Counting
 * Pattern: Bit Manipulation
 *
 * Time Complexity:  O(n) or O(log n)
 * Space Complexity: O(1)
 */

#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    bool divideArray(vector<int>& nums) {
        // Bit manipulation - O(n) time, O(1) space
        int result = 0;
        for (int val : nums) {
            result ^= val;
        }
        return result;
    }
};

File: solutions/problem_2206_divide_array_into_equal_pairs/solution.py
"""
Problem 2206: Divide Array Into Equal Pairs
=========================================
Difficulty: Easy
Tags: Array, Hash Table, Bit Manipulation, Counting
Pattern: Bit Manipulation

Time Complexity:  O(n) or O(log n)
Space Complexity: O(1)
"""

class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        # Bit manipulation - O(n) time, O(1) space
        result = 0
        for val in nums:
            result ^= val
        return result
